src/cr/cube/distributions/wishart.py

```diff
@@ -0,0 +1,135 @@
+# encoding: utf-8
+
+"""The CDF of the largest eigenvalue of a Wishart distribution.
+
+Compute the CDF of the Wishart distribution using the algorithm of
+Chiani (2014). The Wishart CDF is not defined in scipy.stats.wishart, nor in
+tensorflow_probability.distributions.wishart. At the time of writing, the only
+implementation of this CDF is the R package rootWishart. This largely
+follows that, but using scipy implementations rather than Boost, and omitting
+multi-precision support.
+"""
+
+from __future__ import division
+
+import numpy as np
+from scipy.special import gamma, gammainc
+from scipy.linalg import det
+from scipy.constants import pi
+
+try:
+    xrange
+except NameError:
+    xrange = range
+
+
+def wishartCDF(X, n_min, n_max):
+    w_p = np.zeros(X.shape)
+    for i, x in np.ndenumerate(X):
```

Comment:
we could only evaluate the lower triangle and copy. this is still, as far as I can tell, sufficiently fast even doing more than double the work. consider benchmarking.